### PR TITLE
feat: loadAllocs cheatcode (Rebased)

### PR DESCRIPTION
**Pull request by Evalir**
Rebase for #6101 with review comments applied. Thanks @clabby !

### CODE REVIEW

File: crates/cheatcodes/src/evm.rs [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
+use ethers_core::utils::GenesisAccount;
 use ethers_signers::Signer;
 use foundry_evm_core::backend::DatabaseExt;
 use foundry_utils::types::ToAlloy;
 use revm::{
     primitives::{Account, Bytecode, SpecId, KECCAK_EMPTY},
     EVMData,
 };
-use std::collections::HashMap;
+use std::{collections::HashMap, fs::File};
```
Comment (mattsse) [inline crates/cheatcodes/src/evm.rs:14]: this should use foundry's fs bindings

File: crates/evm/core/src/backend/mod.rs [hunk 2]
```diff
@@ -1263,6 +1273,53 @@ impl DatabaseExt for Backend {
...
+    /// Loads the account allocs from the given `allocs` map into the passed [JournaledState].
+    ///
+    /// Returns [Ok] if all accounts were successfully inserted into the journal, [Err] otherwise.
+    fn load_allocs(
+        &mut self,
+        allocs: &HashMap<Address, GenesisAccount>,
+        journaled_state: &mut JournaledState,
+    ) -> Result<(), DatabaseError> {
+        // Loop through all of the allocs defined in the map and commit them to the journal.
+        allocs.iter().try_for_each(|(addr, acc)| {
```
Comment (mattsse) [inline crates/evm/core/src/backend/mod.rs:1285]: I'd prefer a simple loop here

File: crates/cheatcodes/src/evm.rs [hunk 3]
```diff
@@ -62,6 +63,23 @@ impl Cheatcode for loadCall {
     }
 }
 
+impl Cheatcode for loadAllocsCall {
+    fn apply_full<DB: DatabaseExt>(&self, ccx: &mut CheatsCtxt<DB>) -> Result {
+        let Self { pathToAllocsJson } = self;
+
+        // First, load the allocs file from the provided path.
+        let file = File::open(pathToAllocsJson)?;
+        let allocs: HashMap<Address, GenesisAccount> = serde_json::from_reader(file)?;
```
Comment (mattsse) [inline crates/cheatcodes/src/evm.rs:72]: this won't work for the genesis file format, which I think we want to support here
this has nested `allocs: HashMap` objects.

so in order to make this cheatcode work for a genesis.json the user would need to extract that first.
instead we should just support both format here, HashMap and `allocs: HashMap`

File: crates/cheatcodes/src/evm.rs [hunk 4]
```diff
@@ -62,6 +63,23 @@ impl Cheatcode for loadCall {
     }
 }
 
+impl Cheatcode for loadAllocsCall {
+    fn apply_full<DB: DatabaseExt>(&self, ccx: &mut CheatsCtxt<DB>) -> Result {
+        let Self { pathToAllocsJson } = self;
+
+        // First, load the allocs file from the provided path.
+        let file = File::open(pathToAllocsJson)?;
```
Comment (mattsse) [inline crates/cheatcodes/src/evm.rs:71]: can use foundry fs utils read to serde_json::Value first

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/cheatcodes/src/evm.rs (hunk 1 above)
Comment (Evalir) [reply crates/cheatcodes/src/evm.rs]: done in https://github.com/foundry-rs/foundry/pull/6207/commits/1292c7b9856b1391d06a456398ab2417f2831134

File: crates/evm/core/src/backend/mod.rs (hunk 2 above)
Comment (Evalir) [reply crates/evm/core/src/backend/mod.rs]: done in https://github.com/foundry-rs/foundry/pull/6207/commits/1292c7b9856b1391d06a456398ab2417f2831134

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**